File: python-nodes/linear/03_add_tax.py
# Node 3: Add Tax
# Input: order + subtotal  →  Output: order + tax fields

TAX_RATE = 0.08

results = []

for item in _items:
    data = item["json"]
    tax_amount = round(data["subtotal"] * TAX_RATE, 2)
    results.append(
        {
            "json": {
                **data,
                "tax_rate": TAX_RATE,
                "tax_amount": tax_amount,
                "total_before_discount": round(data["subtotal"] + tax_amount, 2),
            }
        }
    )

return results

File: javascript-nodes/linear/03_add_tax.js
// Node 3: Add Tax
// Input: order + subtotal  →  Output: order + tax fields

const TAX_RATE = 0.08;
const results = [];

for (const item of $input.all()) {
  const data = item.json;
  const tax_amount = Math.round(data.subtotal * TAX_RATE * 100) / 100;

  results.push({
    json: {
      ...data,
      tax_rate: TAX_RATE,
      tax_amount,
      total_before_discount: Math.round((data.subtotal + tax_amount) * 100) / 100,
    },
  });
}

return results;
